### ISSUE TITLE
GIO-Warning: Received property Position with type i does not match expected type x in the expected interface

### ISSUE THREAD

**Issue by vn-ki** [acrisci/playerctl#137]
```
$ playerctl --follow metadata

(playerctl:6612): GLib-GIO-WARNING **: 11:18:51.310: Received property Position with type i does not match expected type x in the expected interface
chrome xesam:album               
chrome xesam:artist              https://music.youtube.com
chrome xesam:title               Those Were The Days (Remastered) - YouTube Music
^C

~ took 2s 
$ playerctl --follow metadata

(playerctl:6682): GLib-GIO-WARNING **: 11:18:57.789: Received property Position with type i does not match expected type x in the expected interface
mpv   xesam:url                 file:///home/vn-ki/videos/one-punch-man-2/one-punch-man-2_4.mp4
mpv   mpris:trackid             '/0'
mpv   xesam:title               one-punch-man-2_4.mp4
mpv   mpris:length              1430118000
^C
```

Tried with two different players. Same problem.

**Comment by acrisci** [comment]
What players?

**Comment by vn-ki** [comment]
Google chrome, mpv, KDE connect mpris

**Comment by acrisci** [comment]
This is a bug in those players, but I can go and fix some players.

I can reproduce on Chrome, but I can't fix it because I don't have the source. Sorry.

Mpv doesn't supply mpris on its own. What plugin are you using?

How can I set up kde connect to reproduce the issue?

**Comment by vn-ki** [comment]
> I can reproduce on Chrome, but I can't fix it because I don't have the source. Sorry.

What about chromium? Here's the commit which adds mpris: https://bugs.chromium.org/p/chromium/issues/detail?id=804121#c2

> Mpv doesn't supply mpris on its own. What plugin are you using?

https://github.com/hoyon/mpv-mpris

> How can I set up kde connect to reproduce the issue?

1. Install `kde-connect` and `plasma-browser-integration`. 
2. Make sure the plasma extension is active.

**Comment by acrisci** [comment]
Right now I think the only buggy one is chrome. I've tested mpv-mpris and plasma-browser-integration and they work fine. Do you have chrome running at the same time as you test the other two? I might be getting the signals from chrome somehow at startup with chrome running which would give the warning even if the other players work fine. That's something I should look into.

**Comment by vn-ki** [comment]
> Do you have chrome running at the same time as you test the other two?

Yes. Your guess must be correct.

**Comment by vn-ki** [comment]
@acrisci Chrome is even weirder. When I do `playerctl play` or `playerctl pause` it acts like `playerctl play-pause` with the error mentioned above. It's very annoying.

**Comment by acrisci** [comment]
Yeah it's going to take digging into the chromium code to fix their implementation. Kind of a big project but definitely worth doing.

**Comment by the-c0d3r** [comment]
This issue also happened to me on my spotify native client
OS: Manjaro
Player: Spotify Native
However pause and play still works as expected, but when calling status, it says "stopped" even though it is not. Any ways I can help dig around this issue? Could it be the issue with the Glib? My Glibc version is 2.29-1

**Comment by acrisci** [comment]
@the-c0d3r this issue indicates a bug in the player and has to be fixed there.

If your player is open source, it's within my personal scope to go fix it when I have the time so I leave the issues up here.

If your player is closed source, contact customer support and tell them it's broken.

**Comment by Dersyx** [comment]
I would like to add: I have the same issue as @the-c0d3r, but yet Spotify used to work just fine with `playerctl status` less than a day ago. 

@acrisci, do you have any idea what the bug would be exactly? I'm willing to contact customer support for Spotify, but I don't want to just name a random error message and say, "Here, figure it out!"

**Comment by SHGuy1** [comment]
Same issue here, but now with audacious

**Comment by GwyndolynMarchant** [comment]
Problem is now present with **cmus** as well

`$ playerctl metadata -f '{{position}}'
(playerctl:250576): GLib-GIO-WARNING **: 18:31:50.111: Received property Position with type i does not match expected type x in the expected interface
3416000029`

**Comment by acrisci** [comment]
It works in chromium, mpv, and audacious for me with latest fedora versions.

For an example letter to spotify, see [here](https://github.com/altdesktop/playerctl/blob/master/letter-to-spotify-support.md).
